blaze-client/src/main/scala/org/http4s/client/blaze/Http1Support.scala

```diff
@@ -69,9 +69,9 @@ final private class Http1Support[F[_]](
                 t
               }
             case Left(e) =>
-              Future.failed(e)
+              Future.failed(new ClientFailure(requestKey, addr, e))
```

Comment:
Anything that happens here is still a failure to connect, isn't it?  I'm not sure about the rename, though if we wanted to introduce wrapping elsewhere, we could have a `ConnectionFailure extends ClientFailure`.

Reply:
i think those exceptions might be more generic ... for example main trigger for this was timeout exception, which is not really connection failure - it's failure to process request in due time

Reply:
It wasn't a timeout in establishing a connection?  I'd be surprised to see a request processing timeout here.

Reply:
ah, then i missed the point of what i was trying to achieve and ConnectionFailure is indeed more descriptive ... but i think this PR is still useful ... however i was trying to cover exception mentioned in #3087 which i think is about request processing

Reply:
Yes, I think your enhancements to the exception are still valuable.  I commented on #3087 for what I think is the `TimeoutException` you're trying to clean up.